components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -456,6 +538,129 @@ private ResultSet getIdpQueryResultSet(Connection dbConnection, String sortedOrd
+                                    dbConnection, idPName, idpId, tenantId));
+                            break;
+                        case IdPManagementConstants.IDP_FEDERATED_AUTHENTICATORS:
+                            String defaultAuthenticatorName = resultSet.getString("DEFAULT_AUTHENTICATOR_NAME");
+
+                            // Get federated authenticators.
+                            identityProvider.setFederatedAuthenticatorConfigs(getFederatedAuthenticatorConfigs(
+                                    dbConnection, idPName, identityProvider, tenantId));
+
+                            if (defaultAuthenticatorName != null &&
+                                    identityProvider.getFederatedAuthenticatorConfigs() != null) {
+                                identityProvider.setDefaultAuthenticatorConfig(IdentityApplicationManagementUtil
+                                        .getFederatedAuthenticator(
+                                                identityProvider.getFederatedAuthenticatorConfigs(),
+                                                defaultAuthenticatorName));
+                            }
+                            break;
+                        case IdPManagementConstants.IDP_PROVISIONING:
+                            JustInTimeProvisioningConfig jitProConfig = new JustInTimeProvisioningConfig();
+                            if ((IdPManagementConstants.IS_TRUE_VALUE)
+                                    .equals(resultSet.getString("INBOUND_PROV_ENABLED"))) {
+                                jitProConfig.setProvisioningEnabled(true);
+                            } else {
+                                jitProConfig.setProvisioningEnabled(false);
+                            }
+
+                            jitProConfig
+                                    .setProvisioningUserStore(resultSet.getString("INBOUND_PROV_USER_STORE_ID"));
+                            identityProvider.setJustInTimeProvisioningConfig(jitProConfig);
+
+                            String defaultProvisioningConnectorConfigName =
+                                    resultSet.getString("DEFAULT_PRO_CONNECTOR_NAME");
+                            if (defaultProvisioningConnectorConfigName != null) {
+                                ProvisioningConnectorConfig defaultProConnector = new ProvisioningConnectorConfig();
+                                defaultProConnector.setName(defaultProvisioningConnectorConfigName);
+                                identityProvider.setDefaultProvisioningConnectorConfig(defaultProConnector);
+                            }
+
+                            // Get provisioning connectors.
+                            identityProvider.setProvisioningConnectorConfigs(getProvisioningConnectorConfigs(
+                                    dbConnection, idPName, idpId, tenantId));
+                            break;
```

Comment:
add default case and throw exception

Reply:
Here I haven't added default check since we didn't consider basic attributes cases. I added the default case when we generate query depends on the provided attribute list [1].
[1] https://github.com/wso2/carbon-identity-framework/pull/2821/files#diff-aef45437564b8a5e30cf2197682d6fa6R502

Reply:
if we do like that, what will be the behavior if a user send a basic attribute in requiredAttributes field

Reply:
we won't honour that provided basic attribute, will go as default behaviour flow where the requested attribute is null.